namespaces/validate_test.go

```diff
@@ -0,0 +1,97 @@
+package namespaces
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+)
+
+func TestValidNamespaces(t *testing.T) {
+	for _, testcase := range []struct {
+		name  string
+		input string
+		err   error
+	}{
+		{
+			name:  "Default",
+			input: "default",
+		},
+		{
+			name:  "Hyphen",
+			input: "default-default",
+		},
+		{
+			name:  "DoubleHyphen",
+			input: "default--default",
+		},
+		{
+			name:  "containerD",
+			input: "containerd.io",
+		},
+		{
+			name:  "SwarmKit",
+			input: "swarmkit.docker.io",
+		},
+		{
+			name:  "Punycode",
+			input: "zn--e9.org", // or something like it!
+		},
+		{
+			name:  "LeadingPeriod",
+			input: ".foo..foo",
+			err:   errNamespaceInvalid,
+		},
+		{
+			name:  "Path",
+			input: "foo/foo",
+			err:   errNamespaceInvalid,
+		},
+		{
+			name:  "ParentDir",
+			input: "foo/..",
+			err:   errNamespaceInvalid,
+		},
+		{
+			name:  "RepeatedPeriod",
+			input: "foo..foo",
+			err:   errNamespaceInvalid,
+		},
+		{
+			name:  "OutOfPlaceHyphenEmbedded",
+			input: "foo.-boo",
+			err:   errNamespaceInvalid,
+		},
+		{
+			name:  "OutOfPlaceHyphen",
+			input: "-foo.boo",
+			err:   errNamespaceInvalid,
+		},
+		{
+			name:  "OutOfPlaceHyphenEnd",
+			input: "foo.boo",
+			err:   errNamespaceInvalid,
```

Comment:
> The test should have failed on that case

Yeah, I can't see why it wouldn't. The CI logs have:
```
ok  	github.com/containerd/containerd/namespaces	1.017s	coverage: 61.1% of statements
```
so it appears they did run.